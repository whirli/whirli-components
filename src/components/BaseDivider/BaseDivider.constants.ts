export const PropKeys: Record<string, string> = {
  COLOR: 'color',
  SIZE: 'size',
};

export const PropValues: Record<string, string[]> = {
  [PropKeys.COLOR]: ['default'],
  [PropKeys.SIZE]: ['default'],
};
