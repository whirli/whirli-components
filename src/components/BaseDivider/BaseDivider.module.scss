@import '../../styles/Breakpoints.module.scss';
@import '../../styles/Spacers.scss';

// default styles
.divider {
  border: none;
}

.size-default {
    height: $spacerHairline;
}

.color-default {
    background-color: black;
}

// breakpoint styles
$packageBreakpointStyles: ();