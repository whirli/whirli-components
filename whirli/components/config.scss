@import '@whirli-components/styles/Helpers.scss';

$breakpointPrefix: BREAKPOINT_;

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 672px,
    ),
    small_desk: (
        prefix: sd,
        min-width: 1056px
    ),
    large_desk: (
        prefix: ld,
        min-width: 1312px
    ),
    huge_desk: (
        prefix: hd,
        min-width: 1584px
    ),
);

:export {
    prefix: $breakpointPrefix;
    #{$breakpointPrefix}default: 0px;
    @each $name, $value in $breakpoints {
        #{$breakpointPrefix}#{$name}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}