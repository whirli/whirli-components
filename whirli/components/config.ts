import BaseAside from './BaseAside/BaseAside.config';
import BaseButton from './BaseButton/BaseButton.config';
import BaseColumn from './BaseColumn/BaseColumn.config';
import BaseContainer from './BaseContainer/BaseContainer.config';
import BaseGrid from './BaseGrid/BaseGrid.config';
import BaseImage from './BaseImage/BaseImage.config';
import BaseMain from './BaseMain/BaseMain.config';
import BaseSection from './BaseSection/BaseSection.config';
import BaseSpacer from './BaseSpacer/BaseSpacer.config';
import BaseText from './BaseText/BaseText.config';

export default {
  BaseAside,
  BaseButton,
  BaseColumn,
  BaseContainer,
  BaseGrid,
  BaseImage,
  BaseMain,
  BaseSection,
  BaseSpacer,
  BaseText,
};
