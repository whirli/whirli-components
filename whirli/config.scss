@import '@whirli-components/src/styles/Helpers.scss';

$breakpoints: (
    tablet: (
        prefix: t,
        min-width: 48rem,
    ),
    desktop: (
        prefix: d,
        min-width: 80rem,
    )
);

:export {
    @each $name, $value in $breakpoints {
        BREAKPOINT_#{to-upper-case(#{$name})}: map-deep-get($breakpoints, #{$name}, 'min-width')
    }
}