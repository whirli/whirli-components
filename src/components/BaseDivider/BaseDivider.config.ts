// @ts-ignore
import { PropType } from 'vue';
import * as PropHelpers from '../../helpers/props';
import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli-local/components/BaseDivider/BaseDivider.config';

// Package config
import { PropKeys, PropValues } from './BaseDivider.constants';
import type { PropColor, PropSize } from './BaseDivider.types';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
  [PropKeys.COLOR]: {
    classes: {
      default: 'color-default',
      ...config.styles[PropKeys.COLOR]?.classes,
    } as Record<PropColor, string>,
  },
  [PropKeys.SIZE]: {
    classes: {
      default: 'size-default',
      ...config.styles[PropKeys.SIZE]?.classes,
    } as Record<PropSize, string>,
  },
};

export const ConfigProps: ComponentProps = {
  [PropKeys.COLOR]: {
    required: false,
    type: String as PropType<PropColor>,
    default: 'default',
    ...PropHelpers.stylePropValidation(ConfigStyles, PropKeys.COLOR),
  },
  [PropKeys.SIZE]: {
    required: false,
    type: String as PropType<PropColor>,
    default: 'default',
    ...PropHelpers.stylePropValidation(ConfigStyles, PropKeys.SIZE),
  },
  ...config.props,
};
