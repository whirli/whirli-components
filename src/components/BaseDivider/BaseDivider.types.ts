import { PropKeys, PropValues } from './BaseDivider.constants';

const propColor: string[] = PropValues[PropKeys.COLOR];
const propSize: string[] = PropValues[PropKeys.SIZE];

export type PropColor = typeof propColor[number];
export type PropSize = typeof propSize[number];
