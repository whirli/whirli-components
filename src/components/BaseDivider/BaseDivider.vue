<template>
  <hr :class="classes" />
</template>

<script setup="props" lang="ts">
// Vue
import { defineProps } from 'vue';

// Styles
import styles from '@whirli-local/components/BaseDivider/BaseDivider.module.scss';

// Types
import {
  ComponentStyles as ComponentStylesInterface,
  ComponentProps as ComponentPropsInterface,
} from '../../@types/components';

// Data
import { ConfigStyles, ConfigProps } from './BaseDivider.config';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: ComponentPropsInterface = defineProps(ConfigProps);

// Classes
import useClasses from '../../@use/class';
const { makeClasses } = useClasses();
const classes = [styles.divider, ...makeClasses(ComponentStyles, props, styles)];
</script>
