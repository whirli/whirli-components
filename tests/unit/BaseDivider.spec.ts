import { shallowMount } from '@vue/test-utils';
import { propReturnsClass } from '../helpers';

import BaseDivider from '@/components/BaseDivider/BaseDivider.vue';
import { PropKeys } from '@/components/BaseDivider/BaseDivider.constants';
import { ConfigStyles } from '@/components/BaseDivider/BaseDivider.config';

let wrapper: any;

describe('BaseDivider.vue - Single classes', () => {
  beforeEach(() => {
    wrapper = shallowMount(BaseDivider, {});
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('maps to the correct styles', () => {
    expect(propReturnsClass(wrapper, ConfigStyles, PropKeys.SIZE, 'size-default')).toBe(true);
    expect(propReturnsClass(wrapper, ConfigStyles, PropKeys.COLOR, 'color-default')).toBe(true);
  });
});
