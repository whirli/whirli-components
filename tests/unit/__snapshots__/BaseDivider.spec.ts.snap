// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDivider.vue - Single classes renders the component 1`] = `
<hr
  class=""
/>
`;
